File: Kotlin/conversor.kt
fun decimalParaBinario(decimal: Int): String {
    if(decimal==0) return "0"

    var numero = decimal
    var binario = ""
    while(numero>0){
        binario = (numero % 2).toString() + binario
        numero /= 2
    }
    return binario
}
fun main(){
    print("Digite um numero decimal: ")
    val decimal = readLine()?.toIntOrNull() ?: 0
    println("Binario: ${decimalParaBinario(decimal)}")
}

File: R/conversor.R
decimal_para_binario <- function(decimal) {
  if(decimal==0) return("0")

  binario <- ""
  numero <- decimal
  while(numero>0){
    binario <- paste0(numero %% 2, binario)
    numero <- numero %/% 2
  }
  binario
}
decimal <- as.integer(readline("Digite um numero decimal: "))
cat("Binario:", decimal_para_binario(decimal), "\n")
